### PR TITLE
Make platform-specific headers textual

### PR DESCRIPTION
**Pull request by muxi**
platform-specific headers have to be included in module map as textual headers so that they only get compiled when they are actually included. This resolves the warning of 'umbrella header does not include header'.

Includes changes in gRPC-Core and BoringSSL

### CODE REVIEW

File: templates/include/grpc/module.modulemap.template [hunk 1]
```diff
@@ -1,31 +1,54 @@
...
-                      "include/grpc/impl/codegen/sync_windows.h"]
     out = []
     for lib in libs:
       if lib.name in ("grpc", "gpr"):
         out += lib.get('public_headers', [])
-    out = [f for f in out if f not in excluded_files]
+    out = [f for f in out if f not in textual_headers]
+    out = [hdr.split('/', 2)[2] for hdr in out]
+    return out
+
```
Comment (makdharma) [inline templates/include/grpc/module.modulemap.template]: write comment?

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_fullstack_trickle.BM_PumpStreamServerToClient_Trickle_2M_32k.opt.old: 1


[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by muxi** [comment]
Basic Tests C++ Linux [opt] (internal CI) - #10139 #12428 
Basic Tests Windows [dbg] (internal CI) - #12595 (fixed in master)
Basic Tests Windows [opt] (internal CI) - #12595 (fixed in master)
Interop Cloud-to-Prod Tests (internal CI) - Java and Ruby tests not related
Tsan C++ (internal CI) - not related

**Pull request merged**